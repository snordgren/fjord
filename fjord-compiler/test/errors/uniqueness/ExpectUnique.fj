module errors.uniqueness.ExpectUnique

requiresUnique : Int -* Int
requiresUnique x =
  x

passesShared : Int -> Int
passesShared x =
  requiresUnique x

value : Int 
value = 
  passesShared 3
