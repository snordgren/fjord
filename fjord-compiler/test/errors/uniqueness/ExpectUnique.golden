test/errors/uniqueness/ExpectUnique.fj:9:17:
  |
9 |   requiresUnique x
  |                 ^^^
expression has type ?:Int, expected 1:Int
